lua/telescope/_extensions/file_browser/actions.lua

```diff
@@ -356,14 +356,24 @@ fb_actions.open_file = function(prompt_bufnr)
   actions.close(prompt_bufnr)
 end
 
+--- Goto directory in |builtin.file_browser|.
+---@param prompt_bufnr number: The prompt bufnr
+---@param path string: Directory to go to
+fb_actions.goto_dir = function(prompt_bufnr, path)
```

Comment:
I guess this abstraction generally makes sense, but lua does not have partial functions by default. I'm not opposed to having this, but the docs for the function should at a code example on how to use this for own mappings.

telescope core repo has a lot of code tags for reference.

Reply:
Please can you elaborate? I'm a total noob when it comes to lua, so there are definitely tonnes of nuances I'm not currently aware of.

However, with that said, this isn't a partial function is it? It's just a normal function which takes a path, and sets the finder path to that.

So it's used in the original intent for this PR, which is to call `fb_actions.goto_dir(prompt_bufnr, "~/")` when the user types `"~/"`, and similarly it calls `fb_actions.goto_dir(prompt_bufnr, "/")` when the user types `"/"`.

However, since there seemed to be some duplication in the other two functions below (`fb_actions.goto_parent_dir` and fb_actions.goto_cwd`), instead of having that duplication I just call `fb_actions.goto_dir` with parent dir and the cwd respectively.

Reply:
> Please can you elaborate? I'm a total noob when it comes to lua, so there are definitely tonnes of nuances I'm not currently aware of.

> However, with that said, this isn't a partial function is it? It's just a normal function which takes a path, and sets the finder path to that.

The function couldn't be mapped straightforwardly, because you'd have to wrap it in another function. Hence, it's not really an action as such but more of a helper.

Reply:
Ah ok, I see.